### PR TITLE
refactor: [NODE-1155] Replace `mac_address` with `deterministic_ips`

### PR DESCRIPTION
**Pull request by Bownairo**
Following https://github.com/dfinity/ic/pull/2757 and https://github.com/dfinity/ic/pull/2785, with a fix to MAC generation on HostOS that was caught by hourly.

### CODE REVIEW

**Review by andrewbattat — APPROVED** [review]

File: rs/ic_os/os_tools/hostos_tool/src/main.rs [hunk 1]
```diff
@@ -23,12 +21,12 @@ pub enum Commands {
         output_directory: String,
     },
     GenerateIpv6Address {
-        #[arg(short, long, default_value = "HostOS")]
-        node_type: String,
+        #[arg(short, long, default_value = "0")]
+        node_type: u8,
```
Comment (frankdavid) [inline rs/ic_os/os_tools/hostos_tool/src/main.rs:25]: Does it have a benefit if we pass numbers? It'd be nice to pass strings from the CLI for readability.

File: rs/ic_os/os_tools/hostos_tool/src/main.rs [hunk 2]
```diff
@@ -73,24 +71,15 @@ pub fn main() -> Result<()> {
...
-            let generated_mac = generate_mac_address(
-                &mgmt_mac,
-                deployment_settings.deployment.name.as_str(),
-                &NodeType::HostOS,
+            let mgmt_mac = resolve_mgmt_mac(deployment_settings.deployment.mgmt_mac)?;
+            let generated_mac = calculate_deterministic_mac(
+                mgmt_mac,
+                deployment_settings.deployment.name.parse()?,
+                IpVariant::V6,
+                0x0,
```
Comment (frankdavid) [inline rs/ic_os/os_tools/hostos_tool/src/main.rs:79]: What is 0x0? Not sure if we have a style guide for this, but one option is adding a arg-name comment, like `/* my_arg */ 0x0`. Or if 0x0 has a special meaning (HostOS?), we could extract it into a constant

**Comment by frankdavid** [comment]
Looks good, any chance it could be broken up to smaller PRs (it feels like a lot of things are going on in this PR)? Never mind if it's too much work.

**Review by frankdavid — CHANGES_REQUESTED** [review]

File: rs/ic_os/os_tools/hostos_tool/src/main.rs (hunk 2 above)
Comment (Bownairo) [reply rs/ic_os/os_tools/hostos_tool/src/main.rs]: These `u8`s will also be factored out, but I've added comments in the meantime to keep things clear.

File: rs/ic_os/network/src/lib.rs [hunk 3]
```diff
@@ -1,33 +1,132 @@
...
-    let formatted_mac = FormattedMacAddress::from(&generated_mac);
     generate_systemd_config_files(
         output_directory,
         network_info,
-        Some(&formatted_mac),
+        Some(generated_mac),
         &ipv6_address,
     )
 }
+
```
Comment (Bownairo) [inline rs/ic_os/network/src/lib.rs:33]: The following is relocated and consolidated from the removed `mac_address` crate.

**Comment by Bownairo** [comment]
Hourly: https://github.com/dfinity/ic/actions/runs/12018919399
Nightly: https://github.com/dfinity/ic/actions/runs/12018921007

**Review by nmattia — APPROVED** [review]
Thanks a lot @Bownairo !

**Pull request merged**